### ISSUE TITLE
The default code environment for DDP and single GPU training is inconsistent

### ISSUE THREAD

**Issue by Liu-zhi-chao** [ultralytics/ultralytics#15866]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train, Multi-GPU

### Bug

Custom models added to the Yolov8 source code （path: home/code/yolov8/ultralytics）downloaded from git clone (for example, MyConvBlock modules added to block.py) are only valid for single GPU training. When using DDP, the code defaults to looking for block.py from the ultralytics library in the conda environment (path: anaconda/envs/myenv/lib/python3.10/site-packages/ultralytics/nn/models/block.py), which makes the locally modified code invalid when training with multiple GPUs. At this time, can we only train by modifying the ultralytics library code in the conda environment? Is there any other way to solve this problem? 
Looking forward to your answer, thank you!

### Environment

Setup complete ✅ (128 CPUs, 503.6 GB RAM, 24.5/30.0 GB disk)

OS                  Linux-5.4.0-153-generic-x86_64-with-glibc2.35
Environment         Docker
Python              3.10.14
Install             pip
RAM                 503.60 GB
CPU                 AMD EPYC 7T83 64-Core Processor
CUDA                None

numpy               ✅ 1.26.4<2.0.0,>=1.23.0
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.1>=1.4.1
torch               ✅ 2.3.1+cu118>=1.8.0
torchvision         ✅ 0.18.1+cu118>=0.9.0
tqdm                ✅ 4.66.5>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.5>=2.0.0

### Minimal Reproducible Example

    model.train(data=r"ultralytics/cfg/datasets/mydata.yaml", batch=16,
                epochs=100, project='runs/detect/train=50', name='yolov8-twoHydra-VCM2-CrossHydra-Concat-n-SGD0.01-llvip', 
                amp=False,
                workers=20,
                optimizer='SGD',  # Optimizer
                # cos_lr=True,  # Cosine LR Scheduler
                lr0=0.01,
                device=[0,1]
                )  # 训练

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by Y-T-G** [comment]
This seems more like an installation issue. If you have ultralytics installed in the conda environment, then it will 

Did you install the cloned code with `pip install -e .`?

**Comment by Liu-zhi-chao** [comment]
> This seems more like an installation issue. If you have ultralytics installed in the conda environment, then it will
> 
> Did you install the cloned code with `pip install -e .`?
Thank you for your reply! It was my negligence that caused this problem. I successfully solved it by using `pip install -e .`

**Comment by Liu-zhi-chao** [comment]
> This seems more like an installation issue. If you have ultralytics installed in the conda environment, then it will
> 
> Did you install the cloned code with `pip install -e .`?



Thank you for your reply! It was my negligence that caused this problem. I successfully solved it by using `pip install -e .`

**Comment by glenn-jocher** [comment]
@Liu-zhi-chao great to hear that you resolved the issue! If you have any more questions, feel free to ask.
